Tests/Functional/ArgumentParserWithIgnoredArgs/main.swift

```diff
@@ -0,0 +1,22 @@
+// RUN: %{swiftc} %s -o %T/ArgumentParserWithIgnoredArgs
+// RUN: %T/ArgumentParserWithIgnoredArgs
+
+#if os(macOS)
+    import SwiftXCTest
+#else
+    import XCTest
+#endif
+
+class ArgumentParsingWithIgnoredArgsTestCase: XCTestCase {
+    static var allTests = {
+        return [
+          ("testFail", testFail),
+          ("testSuccess", testSuccess),
+        ]
+    }()
+    func testFail() { XCTFail("failure") }
+    func testSuccess() { }
+}
+
+let arguments = ["main", "--testing-library=xctest", "--testing-library", "xctest", "\(String(reflecting: ArgumentParsingTestCase.self))/testSuccess", "--testing-library",]
+XCTMain([testCase(ArgumentParsingTestCase.allTests)], arguments: arguments)
```

Comment:
Should there be any checks on the output/result from this? Does this test fail (and how) without the product code change?

Reply:
The test checks the exit code of the process, which should be non-zero if argument parsing failed.

That's my read anyway—you know this code better than I do.

Reply:
Yeah, sure, makes sense from re-reading the code.